lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -975,24 +975,30 @@ public function getName()
      */
     protected function doModifyLimitQuery($query, $limit, $offset = null)
     {
-        $limit = (int) $limit;
```

Comment:
Can you please remove the integer casting entirely and just rely on the method signature `integer|null` for both `$limit` and `$offset`? Then just check everywhere against `null`. Also please remove the `$has*` variables as they are unnecessary.